src/overlays/actors/ovl_En_Sb/z_en_sb.c

```diff
@@ -21,44 +36,373 @@ const ActorInit En_Sb_InitVars = {
     (ActorFunc)EnSb_Update,
     (ActorFunc)EnSb_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/EnSb_Init.asm")
+static ColliderCylinderInitType1 sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0xF7CFFFFF, 0x04, 0x08 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    { 30, 40, 0, { 0, 0, 0 } },
+};
+
+static DamageTable sDamageTable = {
+    0x00, 0x00, 0x00, 0xF1, 0xF1, 0xF1, 0x00, 0xF1, 0x00, 0x00, 0x00, 0xF1, 0xF1, 0xF1, 0x00, 0x00,
+    0xF1, 0x00, 0x00, 0xF1, 0x00, 0x00, 0x00, 0xF1, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xF1,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_S8(hintId, 39, ICHAIN_CONTINUE),
+    ICHAIN_U8(targetMode, 2, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 30, ICHAIN_STOP),
+};
+
+static Vec3f sFlamePosOffsets[] = {
+    { 5.0f, 0.0f, 0.0f },
+    { -5.0f, 0.0f, 0.0f },
+    { 0.0f, 0.0f, 5.0f },
+    { 0.0f, 0.0f, -5.0f },
+};
+
+typedef enum {
+    /* 0x00 */ SHELLBLADE_OPEN,
+    /* 0x01 */ SHELLBLADE_WAIT_CLOSED,
+    /* 0x02 */ SHELLBLADE_WAIT_OPEN,
+    /* 0x03 */ SHELLBLADE_LUNGE,
+    /* 0x04 */ SHELLBLADE_BOUNCE
+} ShellbladeState;
```

Comment:
Could you move the enum to the header file?

Reply:
for sure! I didn't know where a good home for it would be, but it makes sense to me